Sources/SourceKitLSP/CMakeLists.txt

```
@@ -41,6 +41,16 @@ target_link_libraries(SourceKitLSP PUBLIC
   SKCore
   SKSwiftPMWorkspace
   SourceKitD
+  SwiftSyntax::SwiftBasicFormat
+  $<TARGET_OBJECTS:SwiftSyntax::SwiftParser>
```

Please don't do this.  This is going to cause problems for other platforms.  There is no guarantee that `SwiftParser` is built for static linking; furthermore this says that SourceKitLSP **VENDS** the SwiftParser interfaces.  This should be a static or dynamic link based upon how SwiftParser was built.

```suggestion
  SwiftSyntax::SwiftParser
```

---

When I use SwiftSyntax::SwiftParser I'm greeted with a lovely configure-time error:

```
CMake Error in Sources/SourceKitLSP/CMakeLists.txt:
  Imported target "SwiftSyntax::SwiftParser" includes non-existent path

    "/Users/rwidmann/swift/build/Ninja-DebugAssert/swift-syntax/swift /Users/rwidmann/swift/swift-syntax/Sources/SwiftParser"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error in Sources/SourceKitLSP/CMakeLists.txt:
  Imported target "SwiftSyntax::SwiftParser" includes non-existent path

    "/Users/rwidmann/swift/build/Ninja-DebugAssert/swift-syntax/swift /Users/rwidmann/swift/swift-syntax/Sources/SwiftParser"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```